### ISSUE TITLE
Accessibility and screenreader users for Windows

### ISSUE THREAD

**Issue by denizsincar29** [warpdotdev/Warp#3847]
### Discord username (optional)

denizsincar29

### Describe the solution you'd like?

```Text
Hello. I am a blind screenreader user that is very interested in programming. We use programs called screenreaders to control our computers. They read all elements on the screen and we use keyboard navigation. I discovered warp but don't have a mac. I am surprised with this fact on FAQ:
"the UI is built on our own rendering layer that leverages lower-level graphics APIs like Metal."
So I want to ask, is warp accessible for screenreader users? I am very excited to try warp on windows.
```

### Is your feature request related to a problem? Please describe.

```Text
The experience shows that the more nativeare guis, the more accessible is the app. Emagine this dialog:
screenreader: hey, OS, what is on this window?
OS: there is a button with lable "send", a text box "message"...
but if the app renders graphics with it's own UI, OS would reply:
hey, screenreader, there is only a picture and some events binded that you can click on some parts of the picture.
and screenreader will say to the blind user: the window is empty. But in fact, there is a very ritch gui with very beautiful buttons and lists, but all is rendered by the program itself, not the gui framework or native gui.
I hope you understood the importance of accessibility.
```

### Additional context

__

### How important is this feature to you?

5 (Can't work without it!)

### Warp Internal (ignore) - linear-label:770f6576-d6c0-4e4f-a259-fc64b5156087

None

**Comment by dannyneira** [comment]
Hey @denizsincar29 Thanks for submitting this! There is a way to set "Concise accessibility announcements" which are less verbose. In the Command palette `CMD-P`, search for "a11y" to toggle the setting. Hope this helps!

Please learn more about accessibility in Warp in our docs. https://docs.warp.dev/features/accessibility

**Comment by denizsincar29** [comment]
@dannyneira but well, if an app is not accessible untill i press ctrl+p and toggle a11y, than how will i toggle it myself using screenreader?

**Comment by dannyneira** [comment]
That's a good point, We could do better about recognizing when the screen reader is enabled and then enable the setting in Warp by default. I'll let the team know and we'll post any updates on this thread.

**Comment by rehakp** [comment]
Hello,
I've just tried Warp on Windows and can say that it is totally inaccessible so that I can't tell, without using a OCR feature, what's on the screen. Hope some progress will occur for Windows as well. But thanks for ongoing effort and that you don't ignore accessibility, that's definitely fine! I have been using Windows Terminal so far and it is perfectly accessible, even letting me see scrollback buffer with just screenreader commands, thanks to the UI Automation a11y layer. When designing a11y for Windows, definitely make use of UIA.

**Comment by sportfloh** [comment]
Hi @ @dannyneira ,
a friend of mine just told me about warp, so I tryed to try it out ... but I can't!
I'm visually impared, so I have some eyesight left, but anyway I need a screen reader (Voice Over, macOS).

1. Why is warp not respecting my OS settings for light/dark mode?
2. Why do I have to enable accessibility? It has to be supported out of the box!
3. Why did you implement your own rendering and not use the stuff the respective OSs are offer? Accessibility would be much easyer to be supported.
4. Why didn't you integrate [AccessKit](https://accesskit.dev/) from the beginning.
5. Why do you invent new way's we screen reader users need to use your app (Voice Over is a well established screen reader and for blind/visually impared people are standards and defaults even more important).

Sorry for the rant but it's very annoying to constantly remind people about these basics!
And the fact that it is almost 1.5 years from when the issue has been opened, is really sad.
For now I'll recommend against using warp.

Cheers

ps. greetings to your management, they did a very bad job!

**Comment by Mohamed-E-Fayed** [comment]
Hi,
 I have just tried warp and enabling a11y as mentioned in a comment above and in accessibility with voiceover page, but nothing took place. It would have been nice to try AI features, but can not because of inaccessible interface.
Hope that get fixed soon.

**Comment by dannyneira** [comment]
Hi Folks, to anyone else interested in this feature, please add a 👍 to the [original post at the top](https://github.com/warpdotdev/Warp/issues/3847#issue-1985007959) to signal that you want this feature, and subscribe if you'd like to be notified.

**Comment by coryj627** [comment]
@dannyneira these comments are regarding bugs, not a feature request. Ensuring 15% of the population can use the application is not a "feature request" its basic definition of done criteria. This isn't a matter of which user group yells the loudest for developer attention. It's basic access for an entire group of people.
